test/e2e/deployment_test.go

```diff
@@ -309,10 +310,19 @@ func createNamespaceScopedUser(t *testing.T, username string, clusterScopedSecre
 	_, err = KubeClientset.RbacV1().RoleBindings(roleBinding.Namespace).Create(context.Background(), &roleBinding, metav1.CreateOptions{})
 	require.NoError(t, err)
 
-	// Retrieve the bearer token from the ServiceAccount
-	token, err := clusterauth.GetServiceAccountBearerToken(KubeClientset, ns.Name, serviceAccountName, time.Second*60)
-	require.NoError(t, err)
-	assert.NotEmpty(t, token)
+	var token string
+
+	// Attempting to patch the ServiceAccount can intermittently fail with 'failed to patch serviceaccount "(...)" with bearer token secret: Operation cannot be fulfilled on serviceaccounts "(...)": the object has been modified; please apply your changes to the latest version and try again'
+	// We thus keep trying for up to 20 seconds.
+	waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 20*time.Second, true, func(context.Context) (done bool, err error) {
+		// Retrieve the bearer token from the ServiceAccount
+		token, err = clusterauth.GetServiceAccountBearerToken(KubeClientset, ns.Name, serviceAccountName, time.Second*60)
+
+		// Success is no error and a real token, otherwise keep trying
+		return (err == nil && token != ""), nil
+	})
+	require.NoError(t, waitErr)
+	require.NotEmpty(t, token)
```

Comment:
Do you know what other process can modify it? It might be a sign of some race condition in ArgoCD, which we may need to fix.

Reply:
My expectation is that its a core Kubernetes controller that happens to modify the `ServiceAccount` at the same time as Argo CD. Resource contention on shared resources is pretty common in cases like this. Retrying/re-reconciling on resource update error is a pretty standard controller/operator pattern.